File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] res={-1,-1};
        
        res[0]=search(nums,target,true);
 
       res[1]=search(nums,target,false);
   
        return res;
        
    }
    
    
  int search(int[] nums,int target,boolean fo)
      {
    int n=nums.length;
        int ans=-1;
        int s=0,e=n-1;
        while(s<=e)
            {
            int mid=s+((e-s)/2);
            
            if(target<nums[mid])e=mid-1;
   
            else if(target>nums[mid])
                {
                s=mid+1;
                
                }
            else{
                ans=mid;
                if(fo) e=mid-1;
                
                else{s=mid+1;}
            }
            
                
          }
      return ans;
            
      
        
        
      }  
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.dart
class Solution {
  List<int> searchRange(List<int> nums, int target) {
    int binarySearchLeft(List<int> nums, int target) {
      int left = 0, right = nums.length - 1;
      while (left <= right) {
        int mid = (left + right) ~/ 2;
        if (nums[mid] >= target) {
          right = mid - 1;
        } else {
          left = mid + 1;
        }
      }
      return left;
    }

    int binarySearchRight(List<int> nums, int target) {
      int left = 0, right = nums.length - 1;
      while (left <= right) {
        int mid = (left + right) ~/ 2;
        if (nums[mid] > target) {
          right = mid - 1;
        } else {
          left = mid + 1;
        }
      }
      return right;
    }

    int leftIndex = binarySearchLeft(nums, target);
    if (leftIndex == nums.length || nums[leftIndex] != target) {
      return [-1, -1];
    }
    return [leftIndex, binarySearchRight(nums, target)];
  }
}